src/server/nonClassEvent/__tests__/nonClassEvent.controller.test.ts

```diff
@@ -99,4 +110,31 @@ describe('NonClassEvent controller', function () {
       );
     });
   });
+  describe('create', function () {
+    it('creates non class parents within an existing area', async function () {
+      const mockArea = rawAreaList[0];
+      mockAreaRepository.findOne.resolves(mockArea);
+
+      await controller.create(createNonClassParent);
+
+      strictEqual(
+        mockNonClassEventService.createWithNonClassEvents.args[0][0].area,
+        mockArea
+      );
+    });
+    it('throws BadRequestException for an invalid area', async function () {
+      mockAreaRepository.findOne.resolves(null);
+
+      try {
+        await controller.create({
+          ...createNonClassParent,
+          area: 'I don\'t exist',
+        });
+        fail('Did not throw');
+      } catch (e) {
+        console.log(e);
+        strictEqual(e.status, 400);
+      }
+    });
```

Comment:
You could also do something like:

```ts
assert.rejects(() => controller.create({...}), BadRequestException);
```

Reply:
Oh, that's much nicer! Fixed in 8bfffe8e03e002357379e7d3b746558efddd80f4.

I don't think I knew about `assert.rejects`